playbooks/roles/caasp4/tasks/_assert_packages.yml

```diff
@@ -0,0 +1,75 @@
+---
+#
+# (c) Copyright 2019 SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+- name: Identify localhost SUSE OS distribution and version
+  set_fact:
+    suse_repo_version: "CAASP40-{{ ansible_distribution }}-{{ ansible_distribution_version | replace('.', '') }}"
+  when:
+    - ansible_distribution | lower == "sles"
+
+- name: Configure repositories for skuba and terraforms
+  become: yes
+  zypper_repository:
+    name: "{{ suse_repo_version }}"
+    repo: "{{ socok8s_repos_to_configure[suse_repo_version] }}"
+    autorefresh: True
+    auto_import_keys: yes
+    state: present
+  when:
+    - suse_repo_version is defined
+    - suse_repo_version in socok8s_repos_to_configure
+
+- name: Install required skuba, terraform packages for OS versions
+  become: yes
+  package:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - skuba
+    - terraform
+  when:
+    - suse_repo_version is defined
+    - suse_repo_version in socok8s_repos_to_configure
+
+- name: Check skuba binary is available
+  command: "skuba version"
+  ignore_errors: yes
```

Comment:
Can you remove the two tasks here?

If using skuba and skuba is not present, a "no such file or directory" will be pretty explicit. And skuba should have been installed anyway on the task above.

Reply:
This is just an assertion before other tasks start using skuba functionality. Similar to what we have in existing shell script logic.

If runner host has opensuse Leap 15.x, then packages would not be installed by this logic, and customer still need to do manual steps. So in a way, assertion failure with clear message is helpful.